lace_grid.py

```diff
@@ -13,131 +13,152 @@
-		# define the stroke style
-		s = {'fill': fill}
-	 
-		# create attributes from style and define path
-		attribs = {'style':simplestyle.formatStyle(s), 
-					'cx':str(x),
-					'cy':str(y),
-					'r':str(r)}
-		
-		# insert path object into current layer
-		inkex.etree.SubElement(self.current_layer, inkex.addNS('circle', 'svg'), attribs)
+    """
+    Create a grid for lace with angle as specified
+    """
+    def __init__(self):
+        """
+        Constructor.
+        Defines the "--centerx" option of the script.
+        """
+        # Call the base class constructor.
+        inkex.Effect.__init__(self)
+
+        # Define float option "--angle" with "-a" shortcut and default value "45.0".
+        self.OptionParser.add_option('-a', '--angle', action='store',
+            type='float', dest='angle', default=45.0, help='Grid Angle')
+        
+        # Define float option "--spacing" with "-s" shortcut and default value "10.0" in mm.
+        self.OptionParser.add_option('-s', '--distance', action='store',
+            type='float', dest='spacing', default=10.0, help='Distance between grid points.')
 
-	def drawGridPoint(self, x, y):
-		self.circle(x, y, self.options.size, self.options.color)
-    										 
-	def draw(self):
-		
-		a = self.options.spacing
-		theta = radians(self.options.angle)
-		
-		hgrid = a*sin(theta);
-		vgrid = a*cos(theta)
-		rows = int(ceil(self.options.height / vgrid))
-		cols = int(ceil(self.options.width  / hgrid))
-		y = 0.0
-		
-		for r in range(rows):
-			x = 0.0
-			if (r % 2 == 1):
-				x += hgrid
-			
-			for c in range(cols/2):
-				self.drawGridPoint(x, y)
-				x += 2.0*hgrid;
-				
-			y += vgrid;
+        # Define float option "--width" with "-w" shortcut and default value "100" in mm.
+        self.OptionParser.add_option('-w', '--width', action='store', 
+            type='float', dest='width', default=100, help='Width of entire ground pattern')
+        
+        # Define float option "--height" with "-l" shortcut and default value "100".
+        self.OptionParser.add_option('-l', '--height', action='store', 
+            type='float', dest='height', default=100, help='Height of entire ground pattern')
+        
+        # Define string option "--units" with "-u" shortcut and default value "mm".
+        self.OptionParser.add_option('-u', '--units', action = 'store',
+            type = 'string', dest = 'units', default = 'mm',
+            help = 'All measurements are in these units.')
+            
+        # Define float option "--pindia" with "-p" shortcut and default value "2".
+        self.OptionParser.add_option('-p', '--pindia', action='store', 
+            type='float', dest='pindia', default=2, help='Diameter of the pins')
+        
+        # Define string option "--pincolor" with "-c" shortcut and default value "Grey".      
+        self.OptionParser.add_option('-c', '--pincolor', action = 'store',
+            type = 'string', dest = 'pincolor', default = -1431655936, # Grey

```

Comment:
This is a legacy from first verion I downloaded. Sorry - I see you had removed the border in a subsequent version.
If the dot color was chosen to be a light one (say yellow) it had the advantage of allowing it to be seen more clearly. But it is more drawing and clearly not needed.
Removing it.